storage/src/main/java/io/atomix/storage/statistics/StorageStatistics.java

```diff
@@ -53,4 +64,32 @@ public long getFreeSpace() {
   public long getTotalSpace() {
     return file.getTotalSpace();
   }
+
+  /**
+   * Returns the amount of free memory remaining.
+   *
+   * @return the amount of free memory remaining
+ * @throws MBeanException
+ * @throws ReflectionException
+ * @throws MalformedObjectNameException
+ * @throws AttributeNotFoundException
+ * @throws InstanceNotFoundException
+   */
+  public long getFreeMemory() throws InstanceNotFoundException, AttributeNotFoundException, MalformedObjectNameException, ReflectionException, MBeanException {
```

Comment:
instead of throwing five types of checked exceptions I think it would make sense to catch in here and return -1 on failure wdyt?

Reply:
we cannot print log here, people will not get the proper exception reason . wdyt?

Reply:
Create the object name objects in the constructor?